From e3333faae25be95b209d0cb4f1165067e24afba8 Mon Sep 17 00:00:00 2001
From: Alfred Chen <cchalpha@gmail.com>
Date: Tue, 19 Mar 2019 10:26:52 +0800
Subject: [PATCH 19/43] bmq: Fix BMQ compile fails for x86-UP, again

Fix issue "Linux 5.0.2 BMQ compile fails for x86-UP" alfredchen/bmq#1
---
 kernel/sched/bmq.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/kernel/sched/bmq.c b/kernel/sched/bmq.c
index a393df2be45f..a430dbe24c96 100644
--- a/kernel/sched/bmq.c
+++ b/kernel/sched/bmq.c
@@ -605,8 +605,9 @@ static inline void dequeue_task(struct task_struct *p, struct rq *rq, int flags)
 		clear_bit(p->bmq_idx, rq->queue.bitmap);
 		update_sched_rq_watermark(rq);
 	}
-	if (1 == --rq->nr_running)
+	--rq->nr_running;
 #ifdef CONFIG_SMP
+	if (1 == rq->nr_running)
 		cpumask_clear_cpu(cpu_of(rq), &sched_rq_pending_mask);
 #endif
 
@@ -632,8 +633,9 @@ static inline void enqueue_task(struct task_struct *p, struct rq *rq, int flags)
 	bmq_add_task(p, &rq->queue, p->bmq_idx);
 	set_bit(p->bmq_idx, rq->queue.bitmap);
 	update_sched_rq_watermark(rq);
-	if (2 == ++rq->nr_running)
+	++rq->nr_running;
 #ifdef CONFIG_SMP
+	if (2 == rq->nr_running)
 		cpumask_set_cpu(cpu_of(rq), &sched_rq_pending_mask);
 #endif
 
-- 
2.21.0.777.g83232e3864

